### PR TITLE
Send an IpcSharedMemory in tex_image_2d and tex_sub_image_2d

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22225)

### CODE REVIEW

**Comment by nox** [comment]
This should be ready to be reviewed.

@bors-servo try

**Comment by nox** [comment]
Unhappy tidy, fixed.

**Review by emilio — APPROVED** [review]
Looks good, with that question. Have you checked any perf numbers out of curiosity? Just wondering what the relative speed is between creating an IPC channel vs. a shared memory thing.

File: components/script/dom/webglrenderingcontext.rs [hunk 1]
```diff
@@ -765,16 +759,9 @@ impl WebGLRenderingContext {
...
-            pixels.size,
-            unpacking_alignment,
-            alpha_treatment,
-            y_axis_treatment,
-            tex_source,
-            pixels.data,
-        );
+        let effective_data_type = self
+            .extension_manager
+            .effective_type(data_type.as_gl_constant());
```
[the remark this answers is not in the document]
Comment (nox) [reply components/script/dom/webglrenderingcontext.rs]: It was like that before, and using it makes everything panic on the GL side.

**Pull request merged**